src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.EcDsa.cs

```diff
@@ -25,5 +28,49 @@ internal static partial class Crypto
         // returns the maximum length of a DER encoded ECDSA signature created with this key.
         [DllImport(Libraries.CryptoNative, EntryPoint = "CryptoNative_EcDsaSize")]
         internal static extern int EcDsaSize(SafeEcKeyHandle ecKey);
+
+        // todo: move this to common cng and openssl
+        [DllImport(Libraries.CryptoNative, EntryPoint = "CryptoNative_LookupNidByOidValue")]
+        private static extern int CryptoNative_LookupNidByOidValue(string oidValue, ref int nid);
```

Comment:
If assume you mean CryptoNative_LookupNidByFriendlyName (not CryptoNative_LookupNidByOidValue). Yes they almost do the same. The difference is LookupNidByFriendlyName also looks at the long name (to be consistent with the internal implementation of our Oid class) but it could be changed to use the definition you refer to which just uses the short name, as we know (for now at least) that there is no different long name for curves